v3/src/data-interactive/handlers/handler-functions.ts

```diff
@@ -0,0 +1,130 @@
+import { ICase } from "../../models/data/data-set-types"
+import { toV2Id, toV3CaseId } from "../../utilities/codap-utils"
+import { t } from "../../utilities/translation/translate"
+import {
+  DICaseValues, DIFullCase, DIResources, DISuccessResult, DIUpdateCase, DIUpdateItemResult, DIValues
+} from "../data-interactive-types"
+import { getCaseRequestResultValues } from "../data-interactive-type-utils"
+import { attrNamesToIds, getCaseValues } from "../data-interactive-utils"
+import { caseNotFoundResult, dataContextNotFoundResult, itemNotFoundResult } from "./di-results"
+
+export function deleteCaseBy(resources: DIResources, aCase?: ICase) {
+  const { dataContext } = resources
+  if (!dataContext) return dataContextNotFoundResult
+  if (!aCase) return caseNotFoundResult
+
+  const pseudoCase = dataContext.pseudoCaseMap.get(aCase.__id__)
+  const cases = pseudoCase?.childCaseIds ?? [aCase.__id__]
+
+  dataContext.applyModelChange(() => {
+    dataContext.removeCases(cases)
+  })
+
+  return { success: true }
```

Comment:
`deleteCaseBy` doesn't return the deleted ids?

Reply:
It just returns `{ success: true }` in v2. It would be easy to add the deleted ids, and it's very unlikely that would break any plugins.

Reply:
Seems like we generally do elsewhere, so consistency would argue that we should do so here as well.